### PR TITLE
[#2162] Fix NSInvalidArgumentException in MSAppCenterUserDefaults

### PR DESCRIPTION
**Pull request by DmitriyKirakosyan**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you check UI tests on the sample app? They are not executed on CI.
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

This PR addresses edge case in `MSAppCenterUserDefaults` when `__NSCFBoolean` is used instead of `NSString` as a key in `NSUserDefaults.dictionaryRepresentation`.

## Related PRs or issues
https://github.com/microsoft/appcenter-sdk-apple/issues/2162

[AB#82990](https://msmobilecenter.visualstudio.com/Mobile-Center/_workitems/edit/82990)

### CODE REVIEW

**Review by Jamminroot — DISMISSED** [review]

File: AppCenter/AppCenter/Internals/Model/Util/MSAppCenterUserDefaults.m [hunk 1]
```diff
@@ -47,7 +47,7 @@ - (void)migrateKeys:(NSDictionary *)migratedKeys forService:(NSString *)service
       NSString *oldKeyPrefix = ((MSUserDefaultsPrefixKey *)oldKey).keyPrefix;
       NSArray *userDefaultsDictionary = [[[NSUserDefaults standardUserDefaults] dictionaryRepresentation] allKeys];
       for (NSString *userDefaultsKey in userDefaultsDictionary) {
-        if ([userDefaultsKey hasPrefix:oldKeyPrefix]) {
+        if ([userDefaultsKey isKindOfClass:[NSString class]] && [userDefaultsKey hasPrefix:oldKeyPrefix]) {
```
Comment (russelarms) [inline AppCenter/AppCenter/Internals/Model/Util/MSAppCenterUserDefaults.m:50]: Can we have a test for this change?

**Review by Jamminroot — APPROVED** [review]

**Review by Jamminroot — DISMISSED** [review]

File: AppCenter/AppCenter/Internals/Model/Util/MSAppCenterUserDefaults.m (hunk 1 above)
Comment (DmitriyKirakosyan) [reply AppCenter/AppCenter/Internals/Model/Util/MSAppCenterUserDefaults.m:50]: Added.

**Review by MatkovIvan — COMMENTED** [review]
Please update changelog file

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## Version 3.3.5 (Under development)
 
+### App Center
+
+* **[Fix]** Fix `NSInvalidArgumentException` when using `__NSCFBoolean` as a key in `NSUserDefaults`.
```
Comment (MatkovIvan) [inline CHANGELOG.md:7]:
```suggestion
* **[Fix]** Fix `NSInvalidArgumentException` when using non-string object as a key in `NSUserDefaults`.
```

File: AppCenter/AppCenterTests/MSUserDefaultsTests.m [hunk 3]
```diff
@@ -4,9 +4,9 @@
 #import "MSAppCenterUserDefaults.h"
 #import "MSAppCenterUserDefaultsPrivate.h"
 #import "MSLoggerInternal.h"
-#import "MSTestFrameworks.h"
 #import "MSUtility.h"
 #import "MSWrapperLogger.h"
+#import <XCTest/XCTest.h>
```
Comment (MatkovIvan) [inline AppCenter/AppCenterTests/MSUserDefaultsTests.m:9]: Why do we need change imports here? This one was already imported in `MSTestFrameworks.h`

Comment (DmitriyKirakosyan) [reply AppCenter/AppCenterTests/MSUserDefaultsTests.m:9]: There was a problem with `XCTAssertNoThrow`. It didn't until I added `#import <XCTest/XCTest.h>`.

Comment (Jamminroot) [reply AppCenter/AppCenterTests/MSUserDefaultsTests.m:9]: XCTAssertNoThrow won't work without it

Comment (DmitriyKirakosyan) [reply AppCenter/AppCenterTests/MSUserDefaultsTests.m]: @MatkovIvan Reverted, seems working fine without it indeed. Please take a look.

**Review by MatkovIvan — APPROVED** [review]

**Review by russelarms — APPROVED** [review]
